### PR TITLE
Silence some warnings.

### PR DESCRIPTION
**Pull request by larsbrinkhoff**
This removes many unused variables, and fixes some printf format issues.  Some things could be handled differently, e.g. maybe `(void *)` isn't a good cast from `pthread_t`, but at least this points to the places that need attention.

### CODE REVIEW

File: chtls.c [hunk 1]
```diff
@@ -677,7 +676,7 @@ void *tls_connector(void *arg)
     int v = 0;
     if ((v = SSL_connect(ssl)) <= 0) {
       fprintf(stderr,"%%%% Error: TLS connect (%s) failed (probably cert problem?)\n", td->tls_name);
-      int err = SSL_get_error(ssl, v);
+      SSL_get_error(ssl, v);
```
Comment (larsbrinkhoff) [inline chtls.c:679]: Maybe not call SSL_get_error at all?

File: chtls.c [hunk 2]
```diff
@@ -1436,7 +1435,7 @@ forward_on_tls(struct chroute *rt, u_short schad, u_short dchad, struct chaos_he
 	 (rt->rt_braddr == 0 && (tlsdest[i].tls_addr == rt->rt_dest))
 	 ||
 	 // multiplexed
-	 is_in_mux_list(dchad, &tlsdest[i].tls_muxed)
+	 is_in_mux_list(dchad, tlsdest[i].tls_muxed)
```
Comment (larsbrinkhoff) [inline chtls.c:1438]: The warning here was that the original code refers to an address of an array.

File: ncp.c [hunk 3]
```diff
@@ -1410,8 +1412,8 @@ wait_for_ack_of_pkt(struct conn *conn, int pknum)
       tn.tv_sec--; 
       tn.tv_usec += 1000000;
     }
-    printf("NCP %s for %p done waiting for EOF ack %#x (%s after %ld.%03d)\n", conn_thread_name(conn), conn, pknum,
-	   timedout != 0 ? "timed out" : "acked", tn.tv_sec, tn.tv_usec/1000);
+    printf("NCP %s for %p done waiting for EOF ack %#x (%s after %ld.%03ld)\n", conn_thread_name(conn), conn, pknum,
+	   timedout != 0 ? "timed out" : "acked", tn.tv_sec, (long)(tn.tv_usec/1000));
```
Comment (larsbrinkhoff) [inline ncp.c:1416]: tv_usec is a `long int` so %ld is more appropriate.  The cast is an extra precaution.

File: ncp.c [hunk 4]
```diff
@@ -1556,7 +1558,7 @@ cancel_conn_threads(struct conn *conn)
   if (pthread_equal(conn->conn_to_net_thread, pthread_self()) ||
       pthread_equal(conn->conn_from_sock_thread, pthread_self())) {
     // conn_to_sock is the main thread, which waits for the others to join - they just exit.
-    if (ncp_debug) printf("%%%% Conn %p thread %p (%s) terminating\n", conn, pthread_self(), conn_thread_name(conn));
+    if (ncp_debug) printf("%%%% Conn %p thread %p (%s) terminating\n", conn, (void *)pthread_self(), conn_thread_name(conn));
```
Comment (larsbrinkhoff) [inline ncp.c:1561]: I'm not sure what the best way to printf a pthread_t is.

File: ncp.c [hunk 5]
```diff
@@ -3522,6 +3517,7 @@ socket_to_conn_handler(void *arg)
     }
     break;
   }
+  return NULL;
```
Comment (larsbrinkhoff) [inline ncp.c:3520]: "Shouldn't get here", but this fixes the warning about reaching the end of the function without returning something.

**Comment by larsbrinkhoff** [comment]
There is also this warning, which I didn't do anything about so far.  It could indicate a potential problem:
```
chip.c: In function ‘chip_input_handle_data’:
chip.c:447:22: warning: ‘srcaddr’ may be used uninitialized in this function
       linktab[srcaddr>>8].pkt_badlen++;
```

File: chtls.c (hunk 1 above)
Comment (bictorv) [reply chtls.c:679]: This can be removed, yes!

File: chtls.c (hunk 2 above)
Comment (bictorv) [reply chtls.c:1438]: Thanks, this was the only warning I saw without -Wall.

File: ncp.c (hunk 4 above)
Comment (bictorv) [reply ncp.c:1568]: This is as good as any, it's just to get the unique identity of the thread.

**Comment by bictorv** [comment]
> There is also this warning, which I didn't do anything about so far. It could indicate a potential problem:
> 
> ```
> chip.c: In function ‘chip_input_handle_data’:
> chip.c:447:22: warning: ‘srcaddr’ may be used uninitialized in this function
>        linktab[srcaddr>>8].pkt_badlen++;
> ```

Indeed a problem. Quick fix: put an ```#if 0`` around ```PTLOCK```...```PTUNLOCKN```. The good fix is to rewrite much more, which I'll save for a rainy day.

**Review by bictorv — APPROVED** [review]
I added some more cleanup. What a great feature -Wall is! :-)

**Comment by larsbrinkhoff** [comment]
> I added some more cleanup. What a great feature -Wall is! :-)

Yes!  Even though some complain about the false positives, I think it
does much more good than harm.

**Pull request merged**